### PR TITLE
[NavigationDrawer] Added a top handle to the drawer

### LINKED ISSUE #5515: [NavigationDrawer] Add a top handle to the header view controller
This was filed as an internal issue. If you are a Googler, please visit [b/118369117](http://b/118369117) for more details.

---
#### Internal data
- Associated internal bug: [b/118369117](http://b/118369117)

### PR DESCRIPTION
**Pull request by yarneo**
**Context:**
The Navigation Drawer should have an affordance to show scrollability in a shape of a top handle.

**The Problem:**
Currently we have no top handle in our drawer to show scrollability.

**The Fix:**
In this PR we are adding a top handle to our drawer. the top handle is hidden by default. when the top handle isn't hidden, it is visible when there is more scrolling to perform for the drawer, and then it fades away as the drawer reaches its full height, or when it goes to full screen.

**Testing:**
Unit Tests + Tested on an iPhone X and iPhone 7 on all examples, portrait and landscape.

**Related Bugs:**
Closes #5515 

**Screenshot**
simulator screen shot - iphone x - 2018-11-05 at 14 22 10

**Video of transition**

### CODE REVIEW

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m [hunk 1]
```diff
@@ -15,8 +15,13 @@
 #import "MDCBottomDrawerPresentationController.h"
 
 #import "MDCBottomDrawerViewController.h"
+#import "MaterialPalettes.h"
 #import "private/MDCBottomDrawerContainerViewController.h"
 
+static CGFloat kTopHandleHeight = 2.f;
```
Comment (codeman7) [inline components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m]: _Tip:_ Please use `kTopHandleHeight = (CGFloat)2.0;`
For this and all below.

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m [hunk 2]
```diff
@@ -81,6 +101,21 @@ - (void)presentationTransitionWillBegin {
 
   [self.containerView addSubview:self.scrimView];
 
+  self.topHandle =
+      [[UIView alloc] initWithFrame:CGRectMake(0, 0, kTopHandleWidth, kTopHandleHeight)];
+  self.topHandle.layer.cornerRadius = kTopHandleHeight * 0.5f;
```
Comment (codeman7) [inline components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m]: Same as above for float.

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m [hunk 3]
```diff
@@ -81,6 +101,21 @@ - (void)presentationTransitionWillBegin {
...
+  self.topHandle.backgroundColor = MDCPalette.greyPalette.tint300;
+  self.topHandle.hidden = self.topHandleHidden;
+  self.topHandle.center = CGPointMake(
```
Comment (codeman7) [inline components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m:109]: Shouldn't we use `CGRectGetMidX()`?

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m [hunk 4]
```diff
@@ -179,6 +221,7 @@ - (void)bottomDrawerContainerViewControllerTopTransitionRatio:
   if ([strongDelegate respondsToSelector:@selector(bottomDrawerTopTransitionRatio:
                                                                   transitionRatio:)]) {
     [strongDelegate bottomDrawerTopTransitionRatio:self transitionRatio:transitionRatio];
+    self.topHandle.alpha = 1.f - transitionRatio;
```
Comment (codeman7) [inline components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m]: Same as above for float.

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h [hunk 5]
```diff
@@ -74,4 +74,10 @@
  */
 @property(nonatomic, weak, nullable) id<MDCBottomDrawerPresentationControllerDelegate> delegate;
 
+/**
+ A Boolean indicating if the top handle of the drawer is hidden or visible.
+ Default is set to YES.
```
Comment (jverkoey) [inline components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h]: From UIKit's docs on "hidden":

> A Boolean value that determines whether the view is hidden.

Consider replacing with:

> A Boolean value that determines whether the top handle of the drawer is hidden.
> Default is YES.

File: components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m [hunk 6]
```diff
@@ -463,4 +464,25 @@ - (void)testBottomDrawerCornersAPIFullScreen {
   XCTAssertEqual(self.drawerViewController.maskLayer.minimumCornerRadius, 3.f);
 }
 
+- (void)testBottomDrawerHandle {
+  // When
+  [self.presentationController presentationTransitionWillBegin];
+  // Then
+  XCTAssertNotNil(self.presentationController.topHandle);
+  XCTAssertEqual(CGRectGetWidth(self.presentationController.topHandle.frame), 24.f);
```
Comment (codeman7) [inline components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m]: Same for floats. Shouldn't we use `XCTAssertWithAccuracy` when comparing floats?

File: components/NavigationDrawer/BUILD [hunk 7]
```diff
@@ -28,6 +28,7 @@ mdc_public_objc_library(
         "UIKit",
     ],
     deps = [
+        "//components/Palettes",
```
Comment (codeman7) [inline components/NavigationDrawer/BUILD:31]: Do we typically depend on Palettes within components?

**Review by codeman7 — APPROVED** [review]

File: components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h (hunk 5 above)
Comment (jverkoey) [reply components/NavigationDrawer/src/MDCBottomDrawerPresentationController.h]: Similarly in MDCBottomDrawerViewController.h

**Review by jverkoey — APPROVED** [review]

File: components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m [hunk 8]
```diff
@@ -463,4 +464,25 @@ - (void)testBottomDrawerCornersAPIFullScreen {
...
+  XCTAssertEqual(CGRectGetHeight(self.presentationController.topHandle.frame), 2.f);
+  XCTAssertEqual(self.presentationController.topHandle.layer.cornerRadius, 1.f);
+  XCTAssertEqual(self.presentationController.topHandle.hidden, YES);
+}
```
Comment (jverkoey) [inline components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m:479]: Can we add a test that verifies the alpha behavior of the handle as it scrolls to the top? If not, please file an issue to explore making it possible to do so.

Comment (yarneo) [reply components/NavigationDrawer/tests/unit/MDCNavigationDrawerScrollViewTests.m:479]: created a follow up bug: https://github.com/material-components/material-components-ios/issues/5636

**Pull request merged**